src/util.rs

```diff
@@ -60,21 +38,37 @@ pub fn get_max_mature_number(rpc_client: &mut CkbRpcClient) -> Result<u64, Strin
         .get_tip_header()
         .map(|header| EpochNumberWithFraction::from_full_value(header.inner.epoch.value()))
         .map_err(|err| err.to_string())?;
-    let tip_epoch_number = tip_epoch.number();
-    if tip_epoch_number < cellbase_maturity.number() {
+
+    let tip_epoch_rational = tip_epoch.to_rational();
+    let cellbase_maturity_rational = cellbase_maturity.to_rational();
+
+    if tip_epoch_rational < cellbase_maturity_rational {
         // No cellbase live cell is mature
         Ok(0)
     } else {
+        let difference = tip_epoch_rational - cellbase_maturity_rational;
+        let rounds_down_difference = difference.clone().into_u256();
+        let difference_delta = difference - rounds_down_difference.clone();
+
+        let epoch_number = u64::from_le_bytes(
+            rounds_down_difference.to_le_bytes()[..8]
+                .try_into()
+                .expect("should be u64"),
+        )
+        .into();
         let max_mature_epoch = rpc_client
-            .get_epoch_by_number((tip_epoch_number - cellbase_maturity.number()).into())
+            .get_epoch_by_number(epoch_number)
             .map_err(|err| err.to_string())?
             .ok_or_else(|| "Can not get epoch less than current epoch number".to_string())?;
-        let start_number = max_mature_epoch.start_number;
-        let length = max_mature_epoch.length;
-        Ok(calc_max_mature_number(
-            tip_epoch,
-            Some((start_number.value(), length.value())),
-            cellbase_maturity,
+
+        let max_mature_block_number = ((difference_delta + U256::one())
+            * U256::from(max_mature_epoch.start_number.value()))
```

Comment:
should be:  `length * difference_delta + start_number`

Reply:
resolved, added an unit test.